### ISSUE TITLE
SELinux: Decide to switch SELinux's default to `enforcing`

### ISSUE THREAD

**Issue by gyptazy** [gardenlinux/gardenlinux#1212]
# Decide to switch SELinux's default to `enforcing`
## General
With finalizing the sub tasks of #1199 we may have a fully functional support of SELinux running in `enforcing` mode. This means, we may switch the default from `permissive` to `enforcing`. However, this could still lead into some issues or more complex integration of new features.

## Task
We should discuss and evaluate this switch.

**Comment by nanory** [comment]
Hi @gyptazy,
after discussing this issue with @gehoern, we decided that we will keep SELinux on permissive mode for now. 

It's good that we were able to get rid of all errors and warnings while enabling SELinux (enforcing mode) and were able to test this on the biggest cloud providers (Azure, GCP and AWS) without any issue but since there is no customer which currently requires this by default and enabling it now would mean that we would also be responsible to keep the rules constantly updated, which might not be worth the effort for now, we will keep SELinux in permissive mode. It doesn't mean one can not enable SELinux in its specific features but we shouldn't enable it by default.

We should have a small discussion in future about assumed efforts for keeping the `gardenlinux-selinux-module` up to date. Some references would probably be helpful, so we should keep an eye on it and do an re-evaluation.

However, it's good to know that we have something in place that would work right out of the box with maybe some minor adjustments to the `gardenlinux-selinux-module` package if required.
